docs(proxy) add documentation for dynamic default sni `*` (#2151)
Kong/docs.konghq.com: app/2.1.x/proxy.md

Valid wildcard positions are `mydomain.*`, `*.mydomain.com`, and `*.www.mydomain.com`.

> A default certificate can be added with Kong configuration:

Reviewer: ```suggestion
A default certificate can be added using the following parameters in Kong configuration:
```

1. [`ssl_cert`](/2.1.x/configuration/#ssl_cert)
2. [`ssl_cert_key`](/2.1.x/configuration/#ssl_cert_key)

> Or by dynamically configuring the default certificate with the SNI of `*`:

Reviewer: ```suggestion
Or, by dynamically configuring the default certificate with an SNI of `*`:
```

```bash
$ curl -i -X POST http://localhost:8001/certificates \
    -F "cert=@/path/to/default-cert.pem" \
    -F "key=@/path/to/default-cert.key" \
    -F "snis=*"
HTTP/1.1 201 Created
...
```

Matching of `snis` respects the following priority:

 1. Exact SNI matching certificate

>  2. Search a certificate by prefix wildcard

Reviewer: ```suggestion
 2. Search for a certificate by prefix wildcard
```

>  3. Search a certificate by suffix wildcard

Reviewer: ```suggestion
 3. Search for a certificate by suffix wildcard
```

>  4. Look for a certificate associated with the SNI `*`

Reviewer: ```suggestion
 4. Search for a certificate associated with the SNI `*`
```
For consistency with the above

>  5. The default certificate on file-system

Reviewer: ```suggestion
 5. The default certificate on the file system
```
